src/plot/plot-mp-coverage.py

```diff
@@ -0,0 +1,237 @@
+
+
+def mk_py(row):
+    if pd.isna(row['spec']):
+        return row['year']
+    else:
+        return str(row['year']) + row['spec']
+
+
+def get_ch(protocol_path):
+    chamber = None
+    spl = protocol_path.split('/')[-1].split('-')
+    if len(spl) == 4:
+        chamber = "ek"
+    else:
+        chamber = spl[3]
+    return chamber
+
+
+def get_baseline(row, baseline_df):
+    #print(row)
+    #print(row['year'], row['chamber'], type(row['year']), type(row['chamber']))
+    y = row['year']
+    c = row['chamber']
+    fdf = baseline_df.loc[(baseline_df['year'] == y) & (baseline_df['chamber'] == c)].copy()
+    fdf.reset_index(inplace=True)
+    #print(fdf.at[0, "n_mps"])
+    return fdf.at[0, "n_mps"]
+
+def plot(df):
+    df['parliament_year'] = df['parliament_year'].apply(lambda x: int(x[:4]))
+    pys = df['parliament_year'].unique()
+    means = [df.loc[df['parliament_year']==py, "ratio"].mean() for py in pys]
+
+    p, a = plt.subplots()
+    a.plot(pys, means)
+    lines = a.get_children()
+    for i, l in enumerate(lines, -len(lines)):
+        l.set_zorder(abs(i))
+    a.axhline(y=1, color='green', linestyle='--', linewidth=1, label='_nolegend_')
+    #a.set_title("Ratio: members of parliament to seats")
+    a.spines['top'].set_visible(False)
+    a.spines['right'].set_visible(False)
+    p.savefig(f"input/plots/LREC/mp-coverage-ratio.pdf", dpi=300, format='pdf')
+
+
+
+
+def main(args):
+    print("checking MP coverage...")
+    baseline_df = pd.read_csv(args.mp_baseline)
+    baseline_df['year'] = baseline_df['year'].apply(lambda x: str(x)[:4])
+    #print(baseline_df)
+    dates = pd.read_csv(args.session_dates, sep=";")
+    dates = dates[~dates['protocol'].isin(skip)]
+    dates = dates[~dates['date'].isin(["2021", "1977"])]
+    #print(dates)
+    if "N_MP" not in dates.columns:
+        dates["N_MP"] = None
+    if "passes_test" not in dates.columns:
+        dates["passes_test"] = None
+    if "almost_passes_test" not in dates.columns:
+        dates['almost_passes_test'] = None
+    if "ratio" not in dates.columns:
+        dates["ratio"] = None
+    if "year" not in dates.columns:
+        dates["year"] = None
+    if "spec" not in dates.columns:
+        dates["spec"] = None
+    if "parliament_year" not in dates.columns:
+        dates["parliament_year"] = None
+    if "chamber" not in dates.columns:
+        dates["chamber"] = None
+    if "baseline_N" not in dates.columns:
+        dates["baseline_N"] = None
+    if "MEPs" not in dates.columns:
+        dates["MEPs"] = None
+
+    dates["year"] = dates["protocol"].apply(lambda x: str(x.split('/')[0][:4]))
+    dates["spec"] = dates["protocol"].apply(lambda x: get_spec(x))
+    dates["parliament_year"] = dates.apply(mk_py, axis=1)
+    dates['chamber'] = dates['protocol'].apply(lambda x: get_ch(x))
+    dates['baseline_N'] = dates.apply(get_baseline, args=(baseline_df,), axis=1)
+
+    print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
```

Comment:
I would really like to use the ```trainerlog``` module, or other proper logging. That makes it easier to turn these off, only show the important ones and see when they are printed out.